src/chopped_graph.hpp

```diff
@@ -0,0 +1,850 @@
+    
+    /// Which chopped segment of the node does this correspond to
+    size_t chop_index(const step_handle_t& step_handle) const;
+    
+    /// Execute a function on each path in the graph. If it returns false, stop
+    /// iteration. Returns true if we finished and false if we stopped early.
+    bool for_each_path_handle_impl(const std::function<bool(const path_handle_t&)>& iteratee) const;
+    
+    /// Execute a function on each step of a handle in any path. If it
+    /// returns false, stop iteration. Returns true if we finished and false if
+    /// we stopped early.
+    bool for_each_step_on_handle_impl(const handle_t& handle,
+                                      const std::function<bool(const step_handle_t&)>& iteratee) const;
+    
+    /// For simplicity, we store the same pointer again with the more general type
+    const PathHandleGraph* path_graph;
+    
+    /// This is nontrivial to compute on the fly, so we'll memoize these values
+    spp::sparse_hash_map<path_handle_t, size_t> chopped_steps_count;
+    
+    
+    /// Make some inherited members accesible without scope resolutions
+    using ChoppedGraph<MaxNodeLength>::get_underlying_handle;
+    using ChoppedGraph<MaxNodeLength>::chop_index;
+    using ChoppedGraph<MaxNodeLength>::chopped_size;
+    using ChoppedGraph<MaxNodeLength>::chopped_index_bit_count;
+    using ChoppedGraph<MaxNodeLength>::chopped_index_mask;
+    using ChoppedGraph<MaxNodeLength>::chopped_handle;
+};
+
+/**
+ * The same as ChoppedPathGraph except with the PathPositionHandleGraph interface added
+ */
+template<uint64_t MaxNodeLength = 1024>
+class ChoppedPathPositionGraph : public ChoppedPathGraph<MaxNodeLength>, virtual public PathPositionHandleGraph {
+public:
+    
+    ChoppedPathPositionGraph(const PathPositionHandleGraph& graph);
+    ChoppedPathPositionGraph() = default;
+    virtual ~ChoppedPathPositionGraph() = default;
+    
+    /// Returns the length of a path measured in bases of sequence.
+    size_t get_path_length(const path_handle_t& path_handle) const;
+    
+    /// Returns the position along the path of the beginning of this step measured in
+    /// bases of sequence. In a circular path, positions start at the step returned by
+    /// path_begin().
+    size_t get_position_of_step(const step_handle_t& step) const;
+    
+    /// Returns the step at this position, measured in bases of sequence starting at
+    /// the step returned by path_begin(). If the position is past the end of the
+    /// path, returns path_end().
+    step_handle_t get_step_at_position(const path_handle_t& path, const size_t& position) const;
+    
+protected:
+    
+    const PathPositionHandleGraph* path_position_graph;
+    
+    using ChoppedPathGraph<MaxNodeLength>::chopped_step_handle;
+    using ChoppedPathGraph<MaxNodeLength>::chop_index;
+    using ChoppedPathGraph<MaxNodeLength>::get_underlying_step_handle;
+};
+
+
```

Comment:
Shouldn't the chopped graph be vectorizable if the underlying graph is?

Reply:
Is that necessary in of the places we want to use it? I'm only imagining using this for constructing mapping indexes and then actually doing mappings. I don't think mpmap or map needs the vectorizable interface, but I'm less familiar with giraffe.